aidb/engine/engine.py

```diff
@@ -42,3 +49,37 @@ def execute(self, query: str, **kwargs):
       raise e
     finally:
       self.__del__()
+
+  async def clear_ml_cache(self, service_name_list = None):
+    '''
+    Clear the cache and output table if the ML model has changed.
+    Delete the tables following the inference services' topological order to maintain integrity during deletion.
+    service_name_list: the name of all the changed services. A list of str or None.
+    If the service name list is not given, the output for all the services will be cleared.
+    '''
+    async with self._sql_engine.begin() as conn:
+      service_ordering = self._config.inference_topological_order
+      if service_name_list is None:
+        service_name_list = [bounded_service.service.name for bounded_service in service_ordering]
+      service_name_list = set(service_name_list)
+      
+      # Get all the services that need to be cleared because of foreign key constraints
+      inference_graph = self._config.inference_graph
```

Comment:
The edge in inference_graph doesn't represent two nodes have foreign key constraints